src/main/java/org/truffleruby/core/kernel/KernelNodes.java

```diff
@@ -1665,6 +1665,7 @@ private static RubyString finishFormat(Node node, int formatLength, BytesResult
 
```

Comment:
@andrykonchin Could you integrate this and add an extra commit removing the two `@ReportPolymorphism` and adding always split on the `@CoreMethod`?
It seems good to also check other cases of pack/unpack/format/sprintf/etc to make sure they are correct too.

Reply:
On second thought, It seems good to keep the `@ReportPolymorphism` added in this PR, because then if someone does:
```ruby
def myformat(format, *args)
  sprintf format, *args
end
myformat "%d", 1
myformat "%f", 3.14
```
it would correctly split, and wouldn't with only the always_split.

But let's always split `SprintfNode` since we know we'll want to split that one anyway, and the earlier the better for startup/warmup/footprint.